components/Redelegate/index.tsx

```diff
@@ -2,28 +2,34 @@ import { bondingManager } from "@lib/api/abis/main/BondingManager";
 import { Button } from "@livepeer/design-system";
 import { useHandleTransaction } from "hooks";
 import { useBondingManagerAddress } from "hooks/useContracts";
-import { useContractWrite, usePrepareContractWrite } from "wagmi";
+import { useWriteContract, useSimulateContract } from "wagmi";
 
 const Index = ({ unbondingLockId, newPosPrev, newPosNext }: any) => {
   const { data: bondingManagerAddress } = useBondingManagerAddress();
 
-  const { config } = usePrepareContractWrite({
+  const { data: config } = useSimulateContract({
     address: bondingManagerAddress,
     abi: bondingManager,
     functionName: "rebondWithHint",
     args: [unbondingLockId, newPosPrev, newPosNext],
   });
-  const { data, isLoading, write, error, isSuccess } = useContractWrite(config);
+  const { data, isPending, writeContract, error, isSuccess } =
+    useWriteContract();
 
-  useHandleTransaction("rebond", data, error, isLoading, isSuccess, {
+  useHandleTransaction("rebond", data, error, isPending, isSuccess, {
     unbondingLockId,
     newPosPrev,
     newPosNext,
   });
 
   return (
     <>
-      <Button variant="primary" size="3" onClick={write} css={{ mr: "$3" }}>
```

Comment:
IMO `Button` should be disabled if `config` is not defined, instead of failing silently. This pattern is found a couple of times in the PR.